### PR TITLE
Implement the API client for retrieving info about remote apps.

### PR DESCRIPTION
**Pull request by frankban**
It's the RemoteApplicationInfo call on the Application facade.

### CODE REVIEW

**Comment by frankban** [comment]
QA:

- `juju bootstrap` with the JUJU_DEV_FEATURE_FLAGS=cross-model feature flag;
- `juju add-model test1`;
- `juju deploy mysql`;
- `juju add-model test2`;
- `juju deploy wordpress`;
- `juju relate wordpress test1.mysql`;
- open the GUI to test2 model and run the following in the console:
  `cb = (err, data) => {console.log(err, data)};` and then
  `app.env.remoteApplicationInfo('test1.mysql', cb)`;
- check that the info provided to cb is sane;
- try to break it: like `app.env.remoteApplicationInfo('foo.no-such', cb)`.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6592/

File: jujugui/static/gui/src/app/store/env/api.js [hunk 1]
```diff
@@ -2577,6 +2577,87 @@ YUI.add('juju-env-api', function(Y) {
...
+          - description: a description for the remote application;
+          - url: the remote application URL;
+          - sourceModel: only populated if the application originates from
+            another model on the same controller rather than via an offer URL;
+          - endpoints: each one having the name, role, interface, limit and
+            scope fields;
+          - icon: the remote application icon contents. Note that this is not a
+            URL, this is the actual icon bytes.
+    */
+    remoteApplicationInfo: function(url, callback) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/store/env/api.js]: Maybe `url` should be `uri` 

/pedantic
;)

Comment (frankban) [inline jujugui/static/gui/src/app/store/env/api.js]: Yeah, but that's called url by Juju so I'd just keep that consistent.

**Comment by hatched** [comment]
Thanks @frankban :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**